tests/api-diff/run.t

```diff
@@ -29,12 +29,71 @@ there should be no diff:
   $ api-diff ref.cmi ref.cmi
   API unchanged!
 
-Adding a value:
+## Different .cmi files for type tests:
+
+Generate a basic `.mli` files with two types and a function
+
+### A file with an additional type:
+
+  $ cat > add_type.mli <<EOF
+  > type t = int
+  > type unused_type = string
+  > type added_t = float
+  > val f : t -> string
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc add_type.mli
+
+Run api-watcher on the two cmi files, there should be a difference
+
+  $ api-diff ref.cmi add_type.cmi
+  API changed!
+
+### A file with a removed type:
+
+  $ cat > remove_type.mli <<EOF
+  > type t = int
+  > val f : t -> string
+  > EOF
+
+We generate the .cmi file
+
+  $ ocamlc remove_type.mli
+
+Run api-watcher on the two cmi files, there should be a difference
+
+The api-diff tool fails to detect this particular change and should be fixed!
```

Comment:
Let's add a `TODO` prefix here and a link to the new issue I've created to track the problem. This will help us remember to fix it and make it easier to locate:

```suggestion
TODO: The api-diff tool fails to detect this particular change and should be fixed!
See https://github.com/NathanReb/ocaml-api-watch/issues/23
```